dashboard/app/controllers/pd/teacher_application_controller.rb

```diff
@@ -13,4 +22,96 @@ def new
       render :submitted
     end
   end
+
+  # GET /pd/teacher_application/manage
+  # Admin only
+  def manage
+    query = params[:q]
+    if query
+      # It's a valid id. Navigate directly to that application.
+      if query =~ /^\d+$/ && Pd::TeacherApplication.exists?(query.to_i)
+        redirect_to action: :edit, teacher_application_id: query
+        return
+      end
+
+      # Search emails
+      @teacher_applications = @teacher_applications.where(
+        'primary_email LIKE ? OR secondary_email LIKE ?',
+        "%#{query}%", "%#{query}%"
```

Comment:
Oh, in that case this is probably fine, particularly for an admin-restricted route.